### PR TITLE
Refactor: Splits spellbook.dm into multiple files

### PR DESCRIPTION
**Pull request by Oyu07**
## What Does This PR Do
 Splits spellbook.dm into new files | Moves boo calls to human_spooked.dm

Spells moved to spellbook_entry.dm
One Use spells/books moved to spellbook_one_use.dm

## Why It's Good For The Game
Better file management

## Testing
1. Spawn wizard
2. Test spells
3. All work

## Changelog
NPFC

### CODE REVIEW

File: code/modules/mob/living/carbon/human/human_mob.dm [hunk 1]
```diff
@@ -2071,9 +2071,32 @@ Eyes need to have significantly high darksight to shine unless the mob has the X
...
+	"You feel a chill run down your spine.",
+	"You think you see a figure in your peripheral vision.",
+	"What was that?",
+	"The hairs stand up on the back of your neck.",
+	"You are filled with a great sadness.",
+	"Something doesn't feel right...",
+	"You feel a presence in the room.",
+	"It feels like someone's standing behind you.",
+))
+*/
```
Comment (Henri215) [inline code/modules/mob/living/carbon/human/human_mob.dm:2099]: Why is this commented?

File: code/modules/mob/dead/observer/observer_base.dm [hunk 2]
```diff
@@ -804,3 +804,42 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
 
 	var/datum/spawners_menu/menu = new /datum/spawners_menu(src)
 	menu.ui_interact(src)
+
+/obj/effect/proc_holder/spell/boo
+	name = "Boo!"
+	desc = "Fuck with the living."
+	selection_deactivated_message	= "<span class='notice'>Your presence will not be known. For now.</span>"
+	selection_activated_message		= "<span class='notice'>You prepare to reach across the veil. <b>Left-click to influence a target!</b></span>"
```
Comment (Henri215) [inline code/modules/mob/dead/observer/observer_base.dm:812]:
```suggestion
	selection_deactivated_message = "<span class='notice'>Your presence will not be known. For now.</span>"
	selection_activated_message = "<span class='notice'>You prepare to reach across the veil. <b>Left-click to influence a target!</b></span>"
```

File: code/modules/mob/living/carbon/human/human_mob.dm (hunk 1 above)
Comment (Oyu07) [reply code/modules/mob/living/carbon/human/human_mob.dm:2099]: Ah, forgot to delete it. that was the straight copy paste from its old place, had it as a reference.

Comment (Oyu07) [reply code/modules/mob/living/carbon/human/human_mob.dm]: 4273cf5

File: code/game/gamemodes/wizard/spells.dm [hunk 3]
```diff
@@ -0,0 +1,739 @@
...
+	var/name = "Entry Name"
+	var/is_ragin_restricted = FALSE // FALSE if this is buyable on ragin mages, TRUE if it's not.
+	var/spell_type = null
+	var/desc = ""
+	var/category = "Offensive"
+	var/cost = 2
+	var/refundable = TRUE
+	var/obj/effect/proc_holder/spell/S = null //Since spellbooks can be used by only one person anyway we can track the actual spell
+	var/buy_word = "Learn"
+	var/limit //used to prevent a spellbook_entry from being bought more than X times with one wizard spellbook
```
Comment (DGamerL) [inline code/game/gamemodes/wizard/spells.dm:11]: Could you convert these comments into autodocced ones?

**Review by DGamerL — CHANGES_REQUESTED** [review]
Okay I have 2 general comments on this:
1. Could you rename the spells file to something like `spellbook_entries.dm`? That way it is clear that it is used for the entries in it, instead of looking like it has spells.
2. Could you also clarify that `spellbook.dm` is for the frontent code of the spellbook (If I understand it correctly), and that if you want to add entries, you should go to x file?

File: code/game/gamemodes/wizard/spells.dm (hunk 3 above)
Comment (Oyu07) [reply code/game/gamemodes/wizard/spells.dm]: 40551a0

**Comment by SteelSlayer** [comment]
I don't like how the spellbook entry datum is in a separate file from it's procs. Which datums and procs moved out of the original file also seems very arbitrary. I don't think this is needed.

**Comment by Oyu07** [comment]
> I don't like how the spellbook entry datum is in a separate file from it's procs. Which datums and procs moved out of the original file also seems very arbitrary. I don't think this is needed.

All I have done is move the spells and that spell specific section out, leaving spellbook with only the code that involves the actual book, unless I missed something that is.

File: code/game/gamemodes/wizard/spellbook.dm
Comment (Sirryan2002) [inline code/game/gamemodes/wizard/spellbook.dm:1]: for readability, I'd definitely just put `/datum/spellbook_entry` and all of its procs into its own file `spellbook_entry.dm`, the base type definition for the spellbook_entry datum should proceed all of its procs in the file

File: code/modules/mob/living/carbon/human/human_mob.dm [hunk 4]
```diff
@@ -2071,7 +2071,18 @@ Eyes need to have significantly high darksight to shine unless the mob has the X
...
-	to_chat(src, "<span class='whisper'>[pick(GLOB.boo_phrases)]</span>")
+	var/list/message_list =	list(
+	"You feel a chill run down your spine.",
+	"You think you see a figure in your peripheral vision.",
+	"What was that?",
+	"The hairs stand up on the back of your neck.",
+	"You are filled with a great sadness.",
+	"Something doesn't feel right...",
+	"You feel a presence in the room.",
+	"It feels like someone's standing behind you."
```
Comment (Sirryan2002) [inline code/modules/mob/living/carbon/human/human_mob.dm:2082]:
```suggestion
		"You feel a chill run down your spine.",
		"You think you see a figure in your peripheral vision.",
		"What was that?",
		"The hairs stand up on the back of your neck.",
		"You are filled with great sadness.",
		"Something doesn't feel right...",
		"You feel a presence in the room.",
		"It feels like someone's standing behind you."
```
indentation + grammar

**Review by Sirryan2002 — CHANGES_REQUESTED** [review]

**Review by Burzah — COMMENTED** [review]
I just have a few things, while we are here.

File: code/game/gamemodes/wizard/spellbook.dm
Comment (Sirryan2002) [inline code/game/gamemodes/wizard/spellbook.dm:1]: Lines 1 through 128 need to be move to `spellbook_entry.dm`

**Review by Sirryan2002 — CHANGES_REQUESTED** [review]

File: code/game/gamemodes/wizard/spellbook.dm
Comment (Oyu07) [reply code/game/gamemodes/wizard/spellbook.dm:1]: [c10bc0a](https://github.com/ParadiseSS13/Paradise/pull/23096/commits/c10bc0a22da242996fa92666bf4c3f8ef2f8b86e)

**Comment by Oyu07** [comment]
Closing for now, will return in the future for a do over, properly this time

**Pull request closed without merging**